src/main/scala/com/wavesplatform/state2/reader/CompositeStateReader.scala

```diff
@@ -67,10 +69,26 @@ class CompositeStateReader(inner: StateReader, blockDiff: BlockDiff) extends Sta
 
   override def resolveAlias(a: Alias): Option[Address] = txDiff.aliases.get(a).orElse(inner.resolveAlias(a))
 
-  override def contractContent(name: String): Option[(Boolean, ByteStr, String)] =
-    txDiff.contracts.get(name)
-      .map(t=>(t._1, t._2.bytes, t._3))
-      .orElse(inner.contractContent(name))
+  override def contractContent(id: ByteStr): Option[(Int, Contract)] =
+    txDiff.contracts.get(id)
+      .map(t => (t._1, t._3))
+      .orElse(inner.contractContent(id))
+
+  override def contractInfo(id: ByteStr): Option[DataEntry] =
+    txDiff.contractDB.get(id)
+      .map(t => DataEntry.fromBytes(t).right.get)
+      .orElse(inner.contractInfo(id))
+
+  override def contractTokens(id: ByteStr): Int = inner.contractTokens(id) + txDiff.contractTokens(id)
+
+  override def tokenInfo(id: ByteStr): Option[DataEntry] = {
+    txDiff.tokenDB.get(id)
+      .map(t => DataEntry.fromBytes(t).right.get)
+      .orElse(inner.tokenInfo(id))
+  }
+
+  override def tokenAccountBalance(id: ByteStr): Long =
+    safeSum(txDiff.tokenAccountBalance(id), inner.tokenAccountBalance(id))
```

Comment:
`tokenAccountBalance` is a Map here, directly use `tokenAccountBalance(id)` will return an error if `id` is not a key of the Map. Should use `tokenAccountBalance.getOrElse(id, 0L)` here

may double check it here.